scripts/star/sort_star_input.py

```diff
@@ -41,14 +41,18 @@ def sort_read_groups(
             after `--outSAMattrRGline`.
 
     Returns:
-        list: A list consisting of the input string split on ' , ',
+        tuple: A list consisting of the input string split on ' , ',
             and then sorted by the read group ID values.
-        list: A sorted list of values to the 'ID' key.
+        tuple: A sorted list of values to the 'ID' key.
     """
     read_groups = [rg for rg in read_groups_string.split(" , ")]
-    rgids = [
-        flags.split(" ")[0].split(":")[1] for flags in read_groups
-    ]  # TODO this breaks if flags are out of expected order
+    rgids = []
+    for rg in read_groups:
+        if "ID" not in rg:
+            raise SystemExit("Read group information must contain 'ID' key")
+        for flag in rg.split(","):
+            if "ID" in flag:
+                rgids.append(flag.split(":")[1])
```

Comment:
@adthrasher I spent _way too long_ trying to write this as a nested list comprehension and then gave up and wrote this nested for loop in 5 seconds 😅